client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java

```diff
@@ -189,28 +156,13 @@ public SortBasedShuffleWriter(
         conf,
         client,
         metrics,
-        executorService,
         sendBufferPool);
   }
 
   private void updatePeakMemoryUsed() {
-    // sorter can be null if this writer is closed
-    if (pipelined) {
-      for (SortBasedPusher pusher : pushers) {
-        if (pusher != null) {
-          long mem = pusher.getPeakMemoryUsedBytes();
-          if (mem > peakMemoryUsedBytes) {
-            peakMemoryUsedBytes = mem;
-          }
-        }
-      }
-    } else {
-      if (currentPusher != null) {
```

Comment:
I don't find the null assignment code, so I think such checking is redundant

Reply:
I think it's safe to remove this null check.